All: Update boilerplate

gabrielschulhof opened the pull request:
Fixes gh-7973

Commit:
Panel: no longer jump up on resize

should not scroll to top in just resize window

Fixes gh-7572
Closes gh-7674

arschmitz committed:
Widgets: Pull in new widgets from ui and remove old widgets

This commit pulls in the new jQuery UI Button, Checkboxradio, controlgroup,
and the jQuery UI Accordion. It also Updates to the newest version of
the widget factory which includes the classes option. This commit includes a
backcompat extension for switching boolean style options to classes option.
Lastely it pulls setting theme out into its own extension.

Commit:
Tabs: Override _isLocal() to account for Ajax nav

We cannot compare hrefs to location.href

Fixes gh-7169
Fixes gh-7725
Closes gh-7727

Commit:
Widgets: Updates based on new UI widgets

jaspermdegroot committed:
All: Updated button and icon classes

Fixes gh-7664

Commit:
Popup: Do not reposition if the popup remains on-screen after a resize

Fixes gh-5523
Fixes gh-7862
Closes gh-7893

Commit:
Build: Create custom CSS for the modules listed in the "modules" option

Closes gh-7892

Commit:
Navigation: Remove fallback for browsers lacking dynamic base support

Do not unconditionally inject base tag

Fixes gh-7331
Fixes gh-5757
Fixes gh-7489
Closes gh-7895

Commit:
Page: Add option enhance

Enhances the page only if the option is on, which, by
default, it is.

Fixes gh-7416
Closes gh-7897

Commit:
Popup: Add tabindex after open, and remove it after close

Fixes gh-7856
Closes gh-7956

Commit:
Navigation and defaults: Remove buttonMarkup-related code

Fixes gh-7834
Closes gh-7914

Commit:
Pagecontainer: Hide loader and release transition lock at page load fail

Fixes gh-3143
Closes gh-7915

Commit:
animationComplete: Fix context for timeout/no support

Fixes gh-7916
Closes gh-7920

Commit:
Page: Rename option "closeButton" to "closeBtn" in dialog extension test

Commit:
Build: Include events.js from jquery.mobile.js

events.js was making it into the build only by virtue of the fact that it was
being pulled by navigation.js. In the process of fixing gh-7805 it became clear
that navigation.js doesn't really need events.js. Thus, the reference to
events.js was removed from navigation.js, thereby orphaning events.js from the
build. This rectifies the problem by explicitly adding events.js to the build.

Fixes gh-7925
Closes gh-7926

Commit:
Popup: pagecontainerbeforechange with reloadPage requires immediate flag

The popup must not prevent default when going back to a page with the same URL
from which it was opened if the reloadPage flag is set. Rather, it must close
immediately, handling the page change event as if the user were navigating to a
different page.

Fixes gh-6119
Closes gh-7929

arschmitz committed:
Build: Update dependencies and add qunit-assert-classes

Commit:
Panel: Functionally test jump-to-top avoidance

Closes gh-7924

Commit:
Selectmenu: Focus button during outgoing page transition

Fixes gh-6578
Closes gh-7941

Commit:
Flipswitch: Prevent default when label is clicked

Fixes gh-6661
Closes gh-7942

Commit:
Revert "Forms: .ui-mini text input missing blue highlight on focus"

This reverts commit ab067cb8147f1b1155db0622fda18b792122b730.

Fixes gh-7991
Re gh-6795

Commit:
Slider: Test style and fix tooltip inconsistencies

Commit:
Selectmenu: Enable/disable via prop, not attr

Fixes gh-7997
Closes gh-7998

Commit:
All: Update boilerplate

Fixes gh-7973

arschmitz commented:
@gabrielschulhof this also needs to include adding the proper file header https://github.com/jquery/jquery-ui/blob/master/ui/selectmenu.js#L1-L18 and removing ` //>>excludeStart("jqmBuildExclude", pragmas.jqmBuildExclude);` the now unused delimiter pragmas.

Commit:
Grid: Use clearfix instead of overflow

Fixes gh-6795
Closes gh-7999

gabrielschulhof commented:
@arschmitz Gotcha!

Commit:
Navigation: Avoid redirect error by redirecting with plain JS

arschmitz committed:
Tests: Update to improve ouput and run speed

Simplifys the runs to just be one per jquery version builds on all branches

arschmitz committed:
Tests: Update to improve ouput and run speed

Simplify travis runs to just be one per jquery version

arschmitz committed:
Transition: Don't assume sync then in tests

Fixes gh-8044
Closes gh-8047
Ref jquery/jquery#1996

Commit:
Popup: Use .done() instead of .then() for prerequisites

Fixes gh-8044

Commit:
Navigation: Do not trigger 'navigate' if popstate is default-prevented

Fixes gh-8045
Re gh-8050

Commit:
Selectmenu: Minor fix in filterable demo for custom selectmenu

Closes gh-8055
Fixes gh-8054

Commit:
Navigation: Navigator updates initialHref until initial popstate arrives

Fixes gh-8056
Closes gh-8051

arschmitz committed:
Build: Split up grunt file into individual modules

Also removes some old unused tasks and does a lot of general clean up
including changing most of the task aliases. In addition this splits
up travis runs and only builds and test demos once.

Closes gh-8058

Commit:
Panel: Stretch external panel to document height

Fixes gh-7983
Closes gh-7988

arschmitz committed:
Build: Add cache directory to .travis.yml

arschmitz committed:
Build: Make excludes work with individual files

Also add some excludes to travis build to deal with travis only failures

arschmitz committed:
Build: Style fixes to comply with JSCS and new style guide

arschmitz committed:
Accordion: Style fixes for jscs

arschmitz committed:
Controlgroup: Style fixes for jscs

arschmitz committed:
Button: Style fixes for jscs

arschmitz committed:
Checkboxradio: Style fixes for jscs

arschmitz committed:
Tabs: Style fixes for jscs

arschmitz committed:
Widget: Style fixes for jscs
